apiserver/internal/handlers/resource/resourcemigration.go

```diff
@@ -227,14 +228,37 @@ func getUploadTarget(
 	return target, internalerrors.Capture(err)
 }
 
-type migratingResourceServiceGetter struct {
-	ctxt httpContext
+type resourceDetails struct {
+	// origin identifies where the resource will come from.
+	origin charmresource.Origin
+	// revision is the charm store revision of the resource.
+	revision int
+	// fingerprint is the SHA-384 checksum for the resource blob.
+	fingerprint charmresource.Fingerprint
+	// size is the size of the resource, in bytes.
+	size int64
 }
 
-func (a *migratingResourceServiceGetter) Resource(r *http.Request) (ResourceService, error) {
-	domainServices, err := a.ctxt.domainServicesDuringMigrationForRequest(r)
+// resourceDetailsFromQuery extracts details about the uploaded resource from
+// the request.
+func resourceDetailsFromQuery(query url.Values) (details resourceDetails, err error) {
```

Comment:
Named returned arguments should be used sparingly and for good reason, this doesn't seem like a good reason.